File: 2020/C#/d01.cs
using System;

namespace _2020
{
    class d01
    {
        public static void solve(){
            Console.WriteLine("Day 01");
            string file = System.IO.File.ReadAllText(@"../input/01.txt");
            int[] line = new int[file.Split('\n').Length];
            double Part1 = 0; double Part2 = 0;
            // Conversion
            for (int i=0; i<line.Length; i++){
                line[i] = Convert.ToInt32(file.Split('\n')[i]);
            } 
            // Main loop
            for (int y=0; y<line.Length; y++){
                for (int x=0; x<line.Length; x++){
                    if (line[y] + line[x] == 2020){
                        Part1 = line[y]*line[x];
                    }
                    // Part 2
                    for (int z=0; z<line.Length; z++){
                        if (line[y] + line[x] + line[z] == 2020){
                            Part2 = line[y]*line[x]*line[z];
                        }
                    }
                }
            }
            Console.WriteLine($"Part 1: {Part1}");
            Console.WriteLine($"Part 2: {Part2}");
        }
    }
}

File: 2020/Python/d01.py
from itertools import combinations

data = open("../input/01.txt").read()
line = [int(x) for x in data.split('\n')]

def solve(n, part):
    comb = list(combinations(line, n))
    for el in comb:
        if sum(el) == 2020:
            print(f"Part 1: {el[0] * el[1]}" if part == 1 else f"Part 2: {el[0] * el[1] * el[2]}")

solve(2, part=1)
solve(3, part=2)